Components/Logger.h

```diff
@@ -0,0 +1,46 @@
+#ifndef LOGGER_H
+#define LOGGER_H
+
+#include <QDebug>
+#include <QtGlobal>
+
+class LogAndReturn {
+  QDebug logger;
+
+ public:
+  LogAndReturn(QtMsgType msgType) : logger(msgType) {}
+  LogAndReturn(const QDebug& log) : logger(log) {}

```

Comment:
QDebug may implement move semantics (ie, you may be able to construct using `QDebug&&` from that `qWarning()` object.

Reply:
Ok: d99691125dcb9f17107fdca705b98c43cd4192fd

Qt seems cool with that.

Reply:
Also here's the definition of `qWarning()`.

```cpp
#define qWarning QMessageLogger(QT_MESSAGELOG_FILE, QT_MESSAGELOG_LINE, QT_MESSAGELOG_FUNC).warning
```